### PR TITLE
DAMD-116: Added check to ensure that catId, objId combination is unique.

### PR DESCRIPTION
**Pull request by hassanxp**
Also included unit test check.

### CODE REVIEW

File: python/pfs/datamodel/pfsConfig.py [hunk 1]
```diff
@@ -210,6 +212,15 @@ def validate(self):
             if matrix.shape != (len(self.fiberId), 2):
                 raise RuntimeError("Wrong shape for %s: %s vs (%d,2)" % (nn, matrix.shape, len(self.fiberId)))
 
+        catObjIdList = [(int(catId), int(objId))
+                        for catId, objId in zip(self.catId, self.objId)]
+        sortedCatObjIdList = dict(Counter(sorted(catObjIdList)))
+        duplicates = [{tup: count} for tup, count in sortedCatObjIdList.items() if count > 1]
```
Comment (PaulPrice) [inline python/pfs/datamodel/pfsConfig.py:218]: A list of dicts of a single element?

File: python/pfs/datamodel/pfsConfig.py [hunk 2]
```diff
@@ -210,6 +212,15 @@ def validate(self):
...
+        if len(duplicates) > 0:
+            raise ValueError(f'design {self.pfsDesignId:#016x} contains duplicate occurrences of'
+                             ' the same (catId, objId) combination. Details below:\n'
+                             f'{{(catId, objId): number of occurrences}}:\n\t {duplicates}')
```
Comment (PaulPrice) [inline python/pfs/datamodel/pfsConfig.py:222]: I would write this block as:
```
counts = Counter(zip(self.catId, self.objId))
if counts.most_common(1)[0][1] > 1:
    duplicates = {tup: count for tup, count in counts.items() if count > 1}
    raise ValueError(...)
```

**Review by PaulPrice — CHANGES_REQUESTED** [review]
Some important changes to make, most especially in the test.

File: tests/test_PfsConfig.py [hunk 3]
```diff
@@ -243,6 +243,16 @@ def extendList(values):
...
                 self.makePfsConfig(**{name: array})
 
+        # Duplicate (catId, objId) combos
+        catId = list(self.catId)
+        objId = list(self.objId)
+        catId[0] = catId[1]
+        objId[0] = objId[1]
+        catId[2] = catId[3]
+        objId[2] = objId[3]
+        # with self.assertRaises(ValueError):
```
[the remark this answers is not in the document]
Comment (hassanxp) [reply tests/test_PfsConfig.py]: Remnant of debugging print output. Restored.

File: python/pfs/datamodel/pfsConfig.py [hunk 4]
```diff
@@ -210,6 +212,15 @@ def validate(self):
             if matrix.shape != (len(self.fiberId), 2):
                 raise RuntimeError("Wrong shape for %s: %s vs (%d,2)" % (nn, matrix.shape, len(self.fiberId)))
 
+        catObjIdList = [(int(catId), int(objId))
+                        for catId, objId in zip(self.catId, self.objId)]
+        sortedCatObjIdList = dict(Counter(sorted(catObjIdList)))
+        duplicates = [{tup: count} for tup, count in sortedCatObjIdList.items() if count > 1]
+        if len(duplicates) > 0:
```
[the remark this answers is not in the document]
Comment (hassanxp) [reply python/pfs/datamodel/pfsConfig.py]: No longer relevant.

File: python/pfs/datamodel/pfsConfig.py [hunk 5]
```diff
@@ -210,6 +212,15 @@ def validate(self):
             if matrix.shape != (len(self.fiberId), 2):
                 raise RuntimeError("Wrong shape for %s: %s vs (%d,2)" % (nn, matrix.shape, len(self.fiberId)))
 
+        catObjIdList = [(int(catId), int(objId))
+                        for catId, objId in zip(self.catId, self.objId)]
```
[the remark this answers is not in the document]
Comment (hassanxp) [reply python/pfs/datamodel/pfsConfig.py]: Agreed. Casting was only needed NumPy values (algorithm was developed based on contents of existing design file. Corrected.

File: python/pfs/datamodel/pfsConfig.py (hunk 1 above)
Comment (hassanxp) [reply python/pfs/datamodel/pfsConfig.py]: No longer relevant.

**Review by PaulPrice — APPROVED** [review]

**Pull request merged**